### ISSUE TITLE
pub(crate) breaking rustfmt parsing

### ISSUE THREAD

**Issue by jonhoo** [rust-lang-nursery/rustfmt#1394]
```rust
pub(crate) fn bar(id: usize) -> () {}
```
is turned into the (invalid) program
```rust
pub(crate) fn bar(crate) fn bar(
                  id: usize)
                  -> () {
}
```

**Comment by alobb** [comment]
Something I have found working on this so far: the arg_str created at [items.rs:1552](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/items.rs#L1552) is

```
crate) fn bar(
                  id: usize
```

even though the result string already has `pub(crate)`, so it seems to me that the bug is in the creation of that argument string (i.e., [items.rs:1703](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/items.rs#L1703)).

**Comment by nrc** [comment]
I suspect the problem is something like:
* we want to find comments between arguments
* for the first argument, we are looking for comments between `(` and `id:`
* in this case there is a `(` in `pub(crate)` which we are not expecting, before the `(` in `bar(`

So the fix is to look for the `pub(...)` case, and when searching for comments, search only the span starting at the end of `pub(...)`, rather than at the start of the function signature.

**Comment by nrc** [comment]
@alobb hi, did you get any further with this? Did the above comment make sense?

**Comment by topecongiro** [comment]
This is fixed in `0.8.1 (a718376)`.
